src/prefect/cli/deployment.py

```diff
@@ -409,8 +420,29 @@ async def create_schedule(
         except ObjectNotFound:
             return exit_with_error(f"Deployment {name!r} not found!")
 
+        num_schedules = len(deployment.schedules)
+        noun = "schedule" if num_schedules == 1 else "schedules"
+
+        if replace and num_schedules > 0:
+            if not assume_yes and not typer.confirm(
+                f"Are you sure you want to replace {num_schedules} {noun} for {name}?"
+            ):
+                return exit_with_error("Schedule replacement cancelled.")
+
+            for existing_schedule in deployment.schedules:
+                try:
+                    await client.delete_deployment_schedule(
```

Comment:
I'm not sure it's better than the `for` loop, but _technically_ we could do an `update_deployment` and set `schedules=[]`. One call, but a heavier endpoint.

Reply:
Oh, good idea, I may come back to that.